### PR TITLE
8309692: Fix -Wconversion warnings in javaClasses

### PR DESCRIPTION
**Pull request by coleenp**
Please review this trivial patch to remove implicit integer conversions in javaClasses.inline.hpp/cpp files.
Tested tier1-4.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8309692](https://bugs.openjdk.org/browse/JDK-8309692): Fix -Wconversion warnings in javaClasses (**Enhancement** - P4)

### Reviewers
 * [Frederic Parain](https://openjdk.org/census#fparain) (@fparain - **Reviewer**)
 * [Matias Saavedra Silva](https://openjdk.org/census#matsaave) (@matias9927 - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/14397/head:pull/14397` \
`$ git checkout pull/14397`

Update a local copy of the PR: \
`$ git checkout pull/14397` \
`$ git pull https://git.openjdk.org/jdk.git pull/14397/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 14397`

View PR using the GUI difftool: \
`$ git pr show -t 14397`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/14397.diff">https://git.openjdk.org/jdk/pull/14397.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/14397#issuecomment-1584823674)

### CODE REVIEW

**Review by fparain — APPROVED** [review]

**Comment by coleenp** [comment]
Thanks Fred. I think you agree it's trivial?

**Review by matias9927 — APPROVED** [review]
Looks good! Seems trivial as well

**Comment by fparain** [comment]
Yes, it is trivial.

**Comment by coleenp** [comment]
Thanks Fred and Matias!
/integrate

File: src/hotspot/share/runtime/globals.hpp [hunk 1]
```diff
@@ -1274,7 +1274,7 @@ const int ObjectAlignmentInBytes = 8;
           "(0.0 means off)")                                                \
           range(0.0, (double)max_intx)                                      \
                                                                             \
-  product(intx, MaxJavaStackTraceDepth, 1024,                               \
```
Comment (dholmes-ora) [inline src/hotspot/share/runtime/globals.hpp:1277]: Seems unrelated to the described issue.

File: src/hotspot/share/classfile/javaClasses.inline.hpp [hunk 2]
```diff
@@ -325,15 +325,15 @@ inline bool java_lang_Module::is_instance(oop obj) {
 inline int Backtrace::merge_bci_and_version(int bci, int version) {
   // only store u2 for version, checking for overflow.
   if (version > USHRT_MAX || version < 0) version = USHRT_MAX;
-  assert((jushort)bci == bci, "bci should be short");
-  return build_int_from_shorts(version, bci);
+  assert((u2)bci == bci, "bci should be short");
+  return build_int_from_shorts((u2)version, (u2)bci);
```
Comment (kimbarrett) [inline src/hotspot/share/classfile/javaClasses.inline.hpp:329]: How about instead, something like
```
return build_int_from_shorts((u2)version, checked_cast<u2>(bci));
```
I guess that loses the explicit message, but seems otherwise clearer to me.

File: src/hotspot/share/classfile/javaClasses.inline.hpp [hunk 3]
```diff
@@ -325,15 +325,15 @@ inline bool java_lang_Module::is_instance(oop obj) {
...
 }
 
 inline int Backtrace::merge_mid_and_cpref(int mid, int cpref) {
   // only store u2 for mid and cpref, checking for overflow.
-  assert((jushort)mid == mid, "mid should be short");
-  assert((jushort)cpref == cpref, "cpref should be short");
-  return build_int_from_shorts(cpref, mid);
+  assert((u2)mid == mid, "mid should be short");
+  assert((u2)cpref == cpref, "cpref should be short");
+  return build_int_from_shorts((u2)cpref, (u2)mid);
```
Comment (kimbarrett) [inline src/hotspot/share/classfile/javaClasses.inline.hpp:336]: Similarly here, could use
```
return build_int_from_shorts(checked_cast<u2>(cpref), checked_cast<u2>(mid));
```

**Comment by kimbarrett** [comment]
Oh well, I didn't notice this had already been integrated.
